bitbots_body_behavior/src/bitbots_body_behavior/actions/go_to_defense_position.py

```diff
@@ -43,19 +43,39 @@ def perform(self, reevaluate=False):
 
         goal_position = (-self.blackboard.world_model.field_length / 2, 0)  # position of the own goal
         ball_position = self.blackboard.world_model.get_ball_position_xy()
+        our_pose = np.array(self.blackboard.world_model.get_current_position())
 
         pose_msg = PoseStamped()
         pose_msg.header.stamp = rospy.Time.now()
         pose_msg.header.frame_id = self.blackboard.map_frame
 
-        if self.mode == "freekick":
+        if self.mode == "freekick_first":
             vector_ball_to_goal = np.array(goal_position) - np.array(ball_position)
             # pos between ball and goal but 1m away from ball
             defense_pos = vector_ball_to_goal / np.linalg.norm(vector_ball_to_goal) * 1 + np.array(ball_position)
             pose_msg.pose.position.x = defense_pos[0]
             pose_msg.pose.position.y = defense_pos[1]
             yaw = math.atan(-vector_ball_to_goal[1] / -vector_ball_to_goal[0])
             pose_msg.pose.orientation = Quaternion(*quaternion_from_euler(0, 0, yaw))
+        elif self.mode == "freekick_second":
+            vector_ball_to_goal = np.array(goal_position) - np.array(ball_position)
+            # pos between ball and goal but 1m away from ball and 1m to the side which is closer to us
+            defense_pos = vector_ball_to_goal / np.linalg.norm(vector_ball_to_goal) * 1 + np.array(ball_position)
+            yaw = math.atan(-vector_ball_to_goal[1] / -vector_ball_to_goal[0])
+
+            # decide on side that is closer
+            pos_1 = np.array([defense_pos[0] + math.sin(yaw) * 1, defense_pos[1] + math.cos(yaw) * 1])
```

Comment:
I am not sure about this. This offset is relative to the `defense_pos` which is already an offset to the ball position. I am also not sure about the + and - yaw value. The offset amount should not depend on `yaw`, because `yaw` can be close to 0 which results in both robots going to the same position. Imo an offset with `yaw + 45°` and `yaw - 45°` would be what we want.

Reply:
if 0 is zero then cos(yaw) is one

Reply:
Oh now I see your approach. Nvm.